### DISCUSSION TITLE
PydanticSerializationError: Unable to serialize unknown type: <class 'numpy.int64'>

### DISCUSSION THREAD

**Discussion by Nevermetyou65** [qdrant/qdrant-client#694] [Q&A]
Hello

I tried to upload points to the collection
here is my code 

```
qdrant = qdrant_client.QdrantClient(
    location=os.getenv("QDRANT_SERVE_ADDRESS"), 
    api_key=os.getenv("QDRANT_SERVE_API_KEY"), 
    port=80,
)
print("constructing points..")
points = [
    PointStruct(
        id=i,
        vector=payload["fn_vectors"],
        payload={k: v for k, v in payload.items() if k != "fn_vectors"},
    )
    for i, payload in enumerate(tmp_dict)
]
print("creating collection...")
collection_name = "mtl_medicine_dev_01"
if qdrant.collection_exists(collection_name):
    qdrant.delete_collection(collection_name)

# Create collection with settings  
qdrant.create_collection(
    collection_name,
    vectors_config=VectorParams(
        size=3072,
        distance=Distance.COSINE,
    ),
)
print("uploading points...")
qdrant.upload_points(
    collection_name=collection_name, 
    points=points,
)
```

My variable `tmp_dict` is a list of dictionaries obtained from the pandas data frame with` .to_dict(orient="records").`
I got this error `PydanticSerializationError: Unable to serialize unknown type: <class 'numpy.int64'>`
This is confusing because I have already checked the data type for my payload and vector. There is no `numpy.int64`

Need help with this

Thanks

**Comment by thebirdgr** [comment]
Hi @Nevermetyou65 did you manage to solve it? I have the same issue even though mine is a list, it says numpy array.

**Comment by octocat20** [comment]
I ran into the same thing before. Pandas can leave `numpy.int64` or `numpy.float64` in the dict even when values look like regular Python numbers.

Qdrant cannot serialize those types, so you need to cast before upload:

```python
import numpy as np

def to_py(v):
    if isinstance(v, np.integer):
        return int(v)
    if isinstance(v, np.floating):
        return float(v)
    if isinstance(v, np.ndarray):
        return v.tolist()
    return v

payload = {k: to_py(v) for k, v in row.items() if k != "fn_vectors"}
```

Also make sure your vector is a plain Python list, not a numpy array.
